### PR TITLE
HPCC-21629 Zap Report email option not clearing after submission

### PR DESCRIPTION
**Pull request by kunalaswani**
Email was not resetting on cancel correctly.

Signed-off-by: Aswani, Kunal <kunal.aswani@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by kunalaswani** [comment]
@GordonSmith please review

File: esp/src/eclwatch/WUDetailsWidget.js [hunk 1]
```diff
@@ -156,18 +156,17 @@ define([
...
             },
 
             _onSubmitDialog: function(){
                 var includeSlaveLogsCheckbox = this.includeSlaveLogsCheckbox.get("checked");
+                var emailCheckbox = this.emailCheckbox.get("checked");
                 if (this.zapForm.validate()) {
                     //WUCreateAndDownloadZAPInfo is not a webservice so relying on form to submit.
                     //Server treats "on" and '' as the same thing.
                     this.includeSlaveLogsCheckbox.set("value", includeSlaveLogsCheckbox ? "on" : "off");
+                    this.emailCheckbox.set("value", emailCheckbox ? "on" : "off");
```
Comment (GordonSmith) [inline esp/src/eclwatch/WUDetailsWidget.js]: I don't think this is needed as "SendMail" is a regular Boolean on the server (and am concerned "off" would be considered "true"?)

File: esp/src/eclwatch/WUDetailsWidget.js [hunk 2]
```diff
@@ -156,18 +156,17 @@ define([
...
                 this.zapDialog.hide();
-                this.emailCheckbox.set("value", "false");
-                this.emailSubject.reset();
-                this.emailBody.reset();
                 this.checkThorLogStatus();
             },
 
             _onSubmitDialog: function(){
                 var includeSlaveLogsCheckbox = this.includeSlaveLogsCheckbox.get("checked");
+                var emailCheckbox = this.emailCheckbox.get("checked");
```
Comment (GordonSmith) [inline esp/src/eclwatch/WUDetailsWidget.js]: See below - but I don't think this is needed?

**Review by GordonSmith — COMMENTED** [review]
@kunalaswani questions...

**Comment by kunalaswani** [comment]
@GordonSmith changes tested and made, confirmed with @wangkx , if option SendEmail is not on form(unchecked), no email sent

File: esp/src/eclwatch/WUDetailsWidget.js [hunk 3]
```diff
@@ -156,18 +156,17 @@ define([
...
                 this.zapDialog.hide();
-                this.emailCheckbox.set("value", "false");
-                this.emailSubject.reset();
-                this.emailBody.reset();
                 this.checkThorLogStatus();
             },
 
             _onSubmitDialog: function(){
                 var includeSlaveLogsCheckbox = this.includeSlaveLogsCheckbox.get("checked");
+                //var emailCheckbox = this.emailCheckbox.get("checked");
```
Comment (GordonSmith) [inline esp/src/eclwatch/WUDetailsWidget.js]: Remove commented code.

File: esp/src/eclwatch/WUDetailsWidget.js [hunk 4]
```diff
@@ -156,18 +156,17 @@ define([
...
                 if (this.zapForm.validate()) {
                     //WUCreateAndDownloadZAPInfo is not a webservice so relying on form to submit.
                     //Server treats "on" and '' as the same thing.
                     this.includeSlaveLogsCheckbox.set("value", includeSlaveLogsCheckbox ? "on" : "off");
+                    //this.emailCheckbox.set("value", emailCheckbox ? "on" : "off");
```
Comment (GordonSmith) [inline esp/src/eclwatch/WUDetailsWidget.js]: Remove commented code.

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/eclwatch/WUDetailsWidget.js (hunk 4 above)
Comment (kunalaswani) [reply esp/src/eclwatch/WUDetailsWidget.js]: removed

File: esp/src/eclwatch/WUDetailsWidget.js (hunk 3 above)
Comment (kunalaswani) [reply esp/src/eclwatch/WUDetailsWidget.js]: removed

**Review by GordonSmith — APPROVED** [review]
Looks good.

**Pull request merged**